File: src/main/java/com/baulsupp/oksocial/location/BestLocation.java
package com.baulsupp.oksocial.location;

import java.io.IOException;
import java.util.Optional;

public class BestLocation implements LocationSource {
  @Override public Optional<Location> read() throws IOException {
    FileLocationSource fls = new FileLocationSource(FileLocationSource.FILE);

    Optional<Location> result = fls.read();

    if (!result.isPresent()) {
      CoreLocationCLI coreLoc = new CoreLocationCLI();
      result = coreLoc.read();
    }

    return result;
  }
}

File: src/main/kotlin/com/baulsupp/oksocial/location/BestLocation.kt
package com.baulsupp.oksocial.location

import com.baulsupp.oksocial.output.OutputHandler
import okhttp3.Response

class BestLocation(val outputHandler: OutputHandler<Response>) : LocationSource {
  suspend override fun read(): Location? = FileLocationSource(FileLocationSource.FILE).read()
    ?: CoreLocationCLI(outputHandler).read()
}
